### PR TITLE
adding folly/2020.08.10.00

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **folly/all**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

My ubuntu server does not have enough buffer I can not see the error.... abusing CCI :speak_no_evil:

### CODE REVIEW

**Comment by prince-chrismc** [comment]
relates to #620

File: recipes/folly/all/patches/0008-find-packages.patch [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ endif()
+ 
+-find_package(Glog MODULE)
++find_package(glog MODULE)
+ set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
+ list(APPEND FOLLY_LINK_LIBRARIES ${GLOG_LIBRARY})
+ list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+ 
+-find_package(LibEvent MODULE REQUIRED)
++find_package(Libevent MODULE REQUIRED)
```
Comment (prince-chrismc) [inline recipes/folly/all/patches/0008-find-packages.patch:30]: facebook/folly#1423

**Comment by prince-chrismc** [comment]
More reasons to hate facebook...

```
/home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.cpp:173:54: error: use of deleted function ‘folly::fibers::SemaphoreBase::Waiter::Waiter(folly::fibers::SemaphoreBase::Waiter&&)’
       : semaphoreWaiter(SemaphoreBase::Waiter(tokens)) {
                                                      ^
In file included from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.cpp:17:0:
/home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.h:46:10: note: ‘folly::fibers::SemaphoreBase::Waiter::Waiter(folly::fibers::SemaphoreBase::Waiter&&)’ is implicitly deleted because the default definition would be ill-formed:
   struct Waiter {
          ^
/home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.h:46:10: error: use of deleted function ‘folly::fibers::Baton::Baton(const folly::fibers::Baton&)’
In file included from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.h:21:0,
                 from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.cpp:17:
/home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/Baton.h:41:7: note: ‘folly::fibers::Baton::Baton(const folly::fibers::Baton&)’ is implicitly deleted because the default definition would be ill-formed:
 class Baton {
       ^
/home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/Baton.h:41:7: error: use of deleted function ‘std::atomic<long int>::atomic(const std::atomic<long int>&)’
In file included from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/SharedMutex.h:23:0,
                 from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/Synchronized.h:32,
                 from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.h:20,
                 from /home/conan/workspace/cci_PR-2535/2/1_0/.conan/data/folly/2020.08.10.00/_/_/build/5dfc09525a43139fc8e5c33014ca2e8c04564cbb/source_subfolder/folly/fibers/SemaphoreBase.cpp:17:
/usr/include/c++/5/atomic:650:7: note: declared here
       atomic(const atomic&) = delete;
       ^
```

**Comment by uilianries** [comment]
Folly changes and brakes compatibility frequently ... good luck

**Comment by prince-chrismc** [comment]
The error is

```
[100%] Linking CXX executable bin/test_package
/usr/bin/ld: /home/conan/workspace/cci_PR-2535/4/4_44/.conan/data/folly/2020.08.10.00/_/_/package/e5b12708addf894c02be8549c6513a85068a9281/lib/libfolly_exception_tracer.so: undefined reference to `folly::exception_tracer::installHandlers()'
```

However [the docs](https://github.com/facebook/folly/blob/master/folly/experimental/exception_tracer/README.md) says

> Link in the (full) exception_tracer library. [...]This is less portable (depends on internal details of GNU's libstdc++)

So I am not sure why the old one worked (maybe this was not installed and linked against in the older version)

**Comment by uilianries** [comment]
> The error is
> 
> ```
> [100%] Linking CXX executable bin/test_package
> /usr/bin/ld: /home/conan/workspace/cci_PR-2535/4/4_44/.conan/data/folly/2020.08.10.00/_/_/package/e5b12708addf894c02be8549c6513a85068a9281/lib/libfolly_exception_tracer.so: undefined reference to `folly::exception_tracer::installHandlers()'
> ```
> 
> However [the docs](https://github.com/facebook/folly/blob/master/folly/experimental/exception_tracer/README.md?rgh-link-date=2020-08-19T13%3A18%3A40Z) says
> 
> > Link in the (full) exception_tracer library. [...]This is less portable (depends on internal details of GNU's libstdc++)
> 
> So I am not sure why the old one worked (maybe this was not installed and linked against in the older version)

@prince-chrismc Like I said, Folly changes, breaks and it's hard to follow. Probably your version is outdated now and if you try the latest release, probably you will have a different error. I did in past: Ignore those configurations incompatible, raise a ConanInvalidConfiguration. Trying to patch Folly won't work because they have new releases every week.

**Comment by prince-chrismc** [comment]
Yeah, this was very painful... I throw for this issue but the great of the versions are available

Hopefully fuxx and proxy won't be this bad

File: recipes/folly/all/conanfile.py [hunk 2]
```diff
@@ -13,22 +13,10 @@ class FollyConan(ConanFile):
     license = "Apache-2.0"
     settings = "os", "arch", "compiler", "build_type"
     options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
+    default_options = {"shared": True, "fPIC": True}
```
Comment (Croydon) [inline recipes/folly/all/conanfile.py:16]: Why did you change it to shared?

Comment (prince-chrismc) [reply recipes/folly/all/conanfile.py:16]: to easily test it ... forgot to revert

Comment (prince-chrismc) [reply recipes/folly/all/conanfile.py:16]:
```suggestion
    default_options = {"shared": False, "fPIC": True}
```

**Comment by prince-chrismc** [comment]
Internal Error

```
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: The "GetOutOfDateItems" task failed unexpectedly. [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018: System.InvalidOperationException: The item metadata "%(FullPath)" cannot be applied to the path "follybenchmark.dir\Release\follybenchmark.tlog\ParallelCustomBuild.command.1.tlog". Path: follybenchmark.dir\Release\follybenchmark.tlog\ParallelCustomBuild.command.1.tlog exceeds the OS max path limit. The fully qualified file name must be less than 260 characters. [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.ErrorUtilities.ThrowInvalidOperation(String resourceName, Object[] args) [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Shared.FileUtilities.ItemSpecModifiers.GetItemSpecModifier(String currentDirectory, String itemSpec, String definingProjectEscaped, String modifier, String& fullPath) [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.Microsoft.Build.Framework.ITaskItem2.GetMetadataValueEscaped(String metadataName) [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.Utilities.TaskItem.GetMetadata(String metadataName) [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.CPPTasks.GetOutOfDateItems.Execute() [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\VC\VCTargets\Microsoft.CppCommon.targets(181,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() [C:\J\w\cci_PR-2535\9\47b41e80-ce58-40fe-a9fb-d832853b5992\.conan\data\folly\2019.10.21.00\_\_\build\9f777a46238708b5b3082e22b8d4579b12c9cfa5\build_subfolder\source_subfolder\folly\follybenchmark.vcxproj]
```

**Comment by uilianries** [comment]
@prince-chrismc Limit path in Windows again ... please, add `short_paths`

File: recipes/folly/all/conanfile.py [hunk 3]
```diff
@@ -16,19 +16,7 @@ class FollyConan(ConanFile):
...
-        "glog/0.4.0",
-        "libevent/2.1.11",
-        "lz4/1.9.2",
-        "openssl/1.1.1d",
-        "zlib/1.2.11",
-        "bzip2/1.0.8",
-        "zstd/1.3.5",
-        "snappy/1.1.7"
-    )
+
```
Comment (prince-chrismc) [inline recipes/folly/all/conanfile.py:19]:
```suggestion
    short_paths = True
```

**Comment by prince-chrismc** [comment]
Internal error?

```
ERROR: Missing binary: bzip2/1.0.8:09e3a4c43285997784a5fcaefa367cb780af2032
bzip2/1.0.8: WARN: Can't find a 'bzip2/1.0.8' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Release, compiler=gcc, compiler.version=6, os=Linux
- Options: build_executable=True, fPIC=True, shared=False
- Dependencies: 
- Requirements: 
- Package ID: 09e3a4c43285997784a5fcaefa367cb780af2032

ERROR: Missing prebuilt package for 'bzip2/1.0.8'
Try to build from sources with "--build=bzip2"
Or read "http://docs.conan.io/en/latest/faq/troubleshooting.html#error-missing-prebuilt-package"
```

**Comment by Croydon** [comment]
Probably yet another case of missing packages // @jgsogo @uilianries

**Comment by prince-chrismc** [comment]
Wow. It worked.Yay!

**Review by uilianries — APPROVED** [review]
Great Job!

**Review by madebr — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

File: recipes/folly/all/conanfile.py [hunk 4]
```diff
@@ -39,49 +27,73 @@ def _source_subfolder(self):
...
         elif self.settings.os == "Macos" and self.options.shared:
             raise ConanInvalidConfiguration("Folly could not be built by apple-clang as shared library")
-        elif self.settings.os == "Windows" and \
-             self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as shared library")
+        elif self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on Windows as a shared library")
+        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "Visual Studio" and \
+                not self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on Windows as a static library")
```
Comment (SpaceIm) [inline recipes/folly/all/conanfile.py:71]: Folly can't be built as static if Windows or Visual Studio only?

Visual Studio is completly dropped (shared and static) in 2020.08.10.00?

Comment (SpaceIm) [reply recipes/folly/all/conanfile.py:71]: Well, I've tested, and MinGW is just not supported.
So Windows support is completly removed in folly recipe starting from 2020.08.10.00 :s

Comment (prince-chrismc) [reply recipes/folly/all/conanfile.py:71]: Yes I had I nightmare trying to sort of the issues. I suspect there'll be another version in the near future that will revive it...

Comment (prince-chrismc) [reply recipes/folly/all/conanfile.py:71]: https://github.com/conan-io/conan-center-index/pull/2535#issuecomment-686847741, I did not re-open the logs but i had looked into locally for a bit and there was no obvious answer.

Comment (Milerius) [reply recipes/folly/all/conanfile.py:71]: I wonder what is the goal to have this package if we cannot use it on Windows, any solutions ?

Comment (prince-chrismc) [reply recipes/folly/all/conanfile.py:71]: If you look at the logs from #3177 

```
C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(417): message : while compiling class template member function 'void folly::SharedMutexImpl<false,void,std::atomic,false,false>::lock_shared(folly::SharedMutexImpl<false,void,std::atomic,false,false>::Token &)' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1362): message : see reference to function template instantiation 'void folly::SharedMutexImpl<false,void,std::atomic,false,false>::lock_shared(folly::SharedMutexImpl<false,void,std::atomic,false,false>::Token &)' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1571): message : see reference to class template instantiation 'folly::SharedMutexImpl<false,void,std::atomic,false,false>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/lang/Align.h(105): message : see reference to class template instantiation 'folly::detail::max_align_t_<long double,double,float,__int64,long,int,short,bool,char,char16_t,char32_t,wchar_t,void *,std::max_align_t>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(632): message : see reference to class template instantiation 'std::chrono::duration<double,std::ratio<1,1>>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(178): message : see reference to class template instantiation 'std::chrono::duration<__int64,std::nano>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.24.28314\include\chrono(610): message : see reference to class template instantiation 'std::chrono::time_point<std::chrono::steady_clock,std::chrono::nanoseconds>' being compiled [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1673,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1698,13): error C2672: 'operator __surrogate_func': no matching overloaded function found [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1698,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1750,7): error C2672: 'operator __surrogate_func': no matching overloaded function found [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/SharedMutex.h(1750,1): error C2783: 'folly::atomic_ref<T> folly::make_atomic_ref_t::operator ()(T &) const': could not deduce template argument for '__formal' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]

C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly/synchronization/AtomicRef.h(142): message : see declaration of 'folly::make_atomic_ref_t::operator ()' [C:\J\w\cci_PR-3177@6\.conan\data\folly\2020.08.10.00\_\_\build\3a2ea9a0e95cfaa9d8c78ca27d653a42d9ab6f49\source_subfolder\folly_base.vcxproj]
```

It's really bad folly code that prevents it from compiling in windows.... The mistake on my part was not putting a better exception message... that version does not compile on windows

It's unfortunate that it takes so much effort to add a new version and I only noticed the one I started on was bad in windows.
